library/src/main/java/com/parse/internal/signpost/basic/HttpURLConnectionProvider.java

```diff
@@ -0,0 +1,56 @@
+package com.parse.internal.signpost.basic;
+
+import java.io.IOException;
+import java.lang.reflect.Method;
+import java.net.HttpURLConnection;
+import java.net.URL;
+
+/**
+ * Singleton class to provide HttpURLConnection, using native HttpURLConnection or OkHttp
+ */
+public class HttpURLConnectionProvider {
```

Comment:
This project has both 4-spaced and 2-spaced java files, so I thought following the style in the same dir might be better.